File: LeetCode/Java Solutions/Append Characters to String to Make Subsequence/append.java
class Solution {
    public int appendCharacters(String s, String t) {
        int m = s.length();
        int n = t.length();
        int i = 0, j = 0;

        while (i < m && j < n) {
            if (s.charAt(i) == t.charAt(j)) {
                j++;
            }
            i++;
        }

        return n - j;
    }
}

File: LeetCode/Python Solutions/Append Characters to String to Make Subsequence/append.py
class Solution:
    def appendCharacters(self, s: str, t: str) -> int:
        m, n = len(s), len(t)
        i, j = 0, 0
        
        while i < m and j < n:
            if s[i] == t[j]:
                j += 1
            i += 1
        
        return n - j
